src/test/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobListerTest.java

```diff
@@ -0,0 +1,101 @@
+package uk.gov.hmcts.reform.blobrouter.services.storage;
+
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.models.BlobItem;
+import com.azure.storage.blob.models.BlobItemProperties;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.blobrouter.config.ServiceConfiguration;
+import uk.gov.hmcts.reform.blobrouter.model.out.BlobInfo;
+
+import java.time.OffsetDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static java.time.OffsetDateTime.now;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static uk.gov.hmcts.reform.blobrouter.util.TimeUtils.toLocalTimeZone;
+
+@ExtendWith(MockitoExtension.class)
+class BlobListerTest {
+
+    @Mock
+    private BlobServiceClient storageClient;
+
+    @Mock
+    private ServiceConfiguration serviceConfiguration;
+
+    @InjectMocks
+    private BlobLister blobLister;
+
+    private OffsetDateTime expectedCreationTime = now().minus(3, ChronoUnit.HOURS);
+
+    @Test
+    void listBlobs() {
+        given(serviceConfiguration.getSourceContainers())
+            .willReturn(Arrays.asList("bulkscan", "cmc", "sscs"));
+
+        BlobContainerClient bulkscanBlobClient = mock(BlobContainerClient.class);
+        BlobContainerClient cmcBlobClient = mock(BlobContainerClient.class);
+        BlobContainerClient sscsBlobClient = mock(BlobContainerClient.class);
+        given(storageClient.getBlobContainerClient("bulkscan")).willReturn(bulkscanBlobClient);
+        given(storageClient.getBlobContainerClient("cmc")).willReturn(cmcBlobClient);
+        given(storageClient.getBlobContainerClient("sscs")).willReturn(sscsBlobClient);
+
+        mockStorageList(
+            bulkscanBlobClient,
+            Stream.of(
+                blob("bulk_scan_file_new", false),
+                blob("bulk_scan_file_stale", true)
+            )
+        );
+
+        mockStorageList(
+            cmcBlobClient,
+            Stream.of(
+                blob("cmc_scan_file_stale_1", false),
+                blob("cmc_scan_file_stale_2", false)
+            )
+        );
+
+        mockStorageList(sscsBlobClient, Stream.empty());
+
+        List<BlobInfo> blobInfos = blobLister.listBlobs(2);
```

Comment:
as commented above, rename class or method name.
`blobLister.listBlobs(2)` does not indicate that this returns stale blobs or blobs that were created more than 2h ago.